### ISSUE TITLE
cli_tests fail when running in parallel on Windows/MSVC

### ISSUE THREAD

**Issue by rrrooommmaaa** [rnpgp/rnp#1394]
## Description

Looks like one of `cli_tests` consistently hangs on 2 cores on MSVC build.

Let's try and run `cli_tests` with `-j1` and others with `-j${CTEST_PARALLEL}`, maybe?
Is there an easy filter to run all tests except those matching the pattern?

**Comment by ni4** [comment]
@rrrooommmaaa Is it really a hang, not a timeout? cli_tests.Encryption test is the huge one, with a lot of key generations and so on.

**Comment by rrrooommmaaa** [comment]
It hangs on various tests, not always on `cli_tests.Encryption`, probably some random cli test becomes a victim.
Sometime https://github.com/rnpgp/rnp/pull/1393/checks?check_run_id=1747921807 `Compression` passes in 369.18 sec, `Encryption` hangs
Sometime https://github.com/rnpgp/rnp/pull/1387/checks?check_run_id=1734829333 `Encryption` passes in 496.60 sec, `Compression` hangs.
Given that timeout is 3000 -- it does look like a hang (probably in botan library again).
@dewyatt Can you imagine what the reason can be?

**Comment by ni4** [comment]
@rrrooommmaaa But it passed all CI when `CTEST_PARALLEL` was introduced?

**Comment by rrrooommmaaa** [comment]
Maybe it passed with old botan library

**Comment by ni4** [comment]
Yeah, looks like some locking occurs, see output at https://github.com/ni4/rnp/pull/12
If Compression test is run alone it takes ~270 seconds to finish.
